<h1>Class NamingExceptionEvent</h1>   <ul class="inheritance"> <li><a href="../../../java/lang/object">java.lang.Object</a></li> <li> <ul class="inheritance"> <li><a href="../../../java/util/eventobject">java.util.EventObject</a></li> <li> <ul class="inheritance"> <li>javax.naming.event.NamingExceptionEvent</li> </ul> </li> </ul> </li> </ul>   <dl> <dt>All Implemented Interfaces:</dt> <dd><a href="../../../java/io/serializable">Serializable</a></dd> </dl>   <pre data-language="java">public class NamingExceptionEvent
extends EventObject</pre> <p>This class represents an event fired when the procedures/processes used to collect information for notifying listeners of <code>NamingEvent</code>s threw a <code>NamingException</code>. This can happen, for example, if the server which the listener is using aborts subsequent to the <code>addNamingListener()</code> call.</p> <dl> <dt>Since:</dt> <dd>1.3</dd> <dt>See Also:</dt> <dd>
<a href="naminglistener#namingExceptionThrown-javax.naming.event.NamingExceptionEvent-"><code>NamingListener.namingExceptionThrown(javax.naming.event.NamingExceptionEvent)</code></a>, <a href="eventcontext"><code>EventContext</code></a>, Serialized Form</dd> </dl>       <h2 id="field.summary">Fields</h2>  <h2 id="fields.inherited.from.class.java.util.EventObject">Fields inherited from class java.util.<a href="../../../java/util/eventobject">EventObject</a>
</h2> <code><a href="../../../java/util/eventobject#source">source</a></code>    <h2 id="constructor.detail">Constructors</h2>   <h3 id="NamingExceptionEvent-javax.naming.event.EventContext-javax.naming.NamingException-">NamingExceptionEvent</h3> <pre data-language="java">public NamingExceptionEvent(EventContext source,
                            NamingException exc)</pre> <p>Constructs an instance of <code>NamingExceptionEvent</code> using the context in which the <code>NamingException</code> was thrown and the exception that was thrown.</p> <dl> <dt>Parameters:</dt> <dd>
<code>source</code> - The non-null context in which the exception was thrown.</dd> <dd>
<code>exc</code> - The non-null <code>NamingException</code> that was thrown.</dd> </dl>     <h2 id="method.detail">Methods</h2>   <h3 id="getException--">getException</h3> <pre data-language="java">public NamingException getException()</pre> <p>Retrieves the exception that was thrown.</p> <dl> <dt>Returns:</dt> <dd>The exception that was thrown.</dd> </dl>    <h3 id="getEventContext--">getEventContext</h3> <pre data-language="java">public EventContext getEventContext()</pre> <p>Retrieves the <code>EventContext</code> that fired this event. This returns the same object as <code>EventObject.getSource()</code>.</p> <dl> <dt>Returns:</dt> <dd>The non-null <code>EventContext</code> that fired this event.</dd> </dl>    <h3 id="dispatch-javax.naming.event.NamingListener-">dispatch</h3> <pre data-language="java">public void dispatch(NamingListener listener)</pre> <p>Invokes the <code>namingExceptionThrown()</code> method on a listener using this event.</p> <dl> <dt>Parameters:</dt> <dd>
<code>listener</code> - The non-null naming listener on which to invoke the method.</dd> </dl>     <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993&ndash;2017, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    
  </p>
</div>
